freud/data.py

```diff
@@ -0,0 +1,163 @@
+
+    This class represents the unit cell of a crystal structure, which is
+    defined by a lattice and a basis. It provides the basic attributes of the
+    unit cell as well as enabling the generation of systems of points
+    (optionally with some noise) from the unit cell.
+
+    Args:
+        box:
+            A box-like object (see :meth:`~freud.box.Box.from_box`) containing
+            the lattice vectors of the unit cell.
+        basis_positions ((:math:`N_{points}`) :class:`numpy.ndarray`):
+            The basis of the unit cell in fractional coordinates
+            (Default value = ``[[0, 0, 0]]``.
+    """
+    def __init__(self, box, basis_positions=[[0, 0, 0]]):
+        self._box = freud.box.Box.from_box(box)
+        self._basis_positions = np.dot(basis_positions, self.box.to_matrix().T)
+
+    def to_system(self, num_replicas=1, scale=1, sigma_noise=0):
+        """Generate a system from the unit cell.
+
+        Args:
+            num_replicas (:class:`tuple` or `int`):
+                If provided as a single number, the number of replicas in all
+                dimensions. If a tuple, the number of times replicated in each
+                dimension. Must be of the form (nx, ny, 1) for 2D boxes
+                (Default value = 1).
+            scale (float): Amount to scale the unit cell by (in distance units)
+                (Default value = 1).
+            sigma_noise (float):
+                The standard deviation of the normal distribution used to add
+                noise to the positions in the system (Default value = 0).
+
+        Returns:
+            tuple (:class:`freud.box.Box`, :class:`np.ndarray`):
+                A system-like object (see
+                :class:`~freud.locality.NeighborQuery.from_system`.).
+        """
+        try:
+            nx, ny, nz = num_replicas
+        except TypeError:
+            nx = ny = num_replicas
+            nz = 1 if self.box.is2D else num_replicas
+
+        if self.box.is2D and nz != 1:
+            raise ValueError("The number of replicas in z must be 1 for a "
+                             "2D unit cell.")
+
+        basis = np.tile(
+            self.basis_positions[np.newaxis, np.newaxis, np.newaxis, ...],
+            (nx, ny, nz, 1, 1))
+        basis[..., 0] += np.arange(nx)[:, np.newaxis, np.newaxis, np.newaxis]
+        basis[..., 1] += np.arange(ny)[np.newaxis, :, np.newaxis, np.newaxis]
+        basis[..., 2] += np.arange(nz)[np.newaxis, np.newaxis, :, np.newaxis]
+
+        positions = (basis - 0.5).reshape(-1, 3).dot(
+            self.box.to_matrix().T)*scale
+        box = freud.box.Box.from_matrix(
+            self.box.to_matrix() * scale * [[nx, ny, nz]])
+
+        if sigma_noise != 0:
+            mean = [0]*3
+            var = sigma_noise*sigma_noise
+            cov = np.diag([var, var, var if self.dimensions == 3 else 0])
+            positions += np.random.multivariate_normal(
+                mean, cov, size=positions.shape[:-1])
+
```

Comment:
This needs a final step: `positions = box.wrap(positions)` to account for noise pushing particles over the boundaries.